daemon/fqdn.go

```diff
@@ -73,11 +76,88 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState, preCache
 		OverLimit:      option.Config.ToFQDNsMaxIPsPerHost,
 		Cache:          fqdn.NewDNSCache(option.Config.ToFQDNsMinTTL),
 		LookupDNSNames: fqdn.DNSLookupDefaultResolver,
-		AddGeneratedRules: func(generatedRules []*policyApi.Rule) error {
+		AddGeneratedRulesAndUpdateSelectors: func(generatedRules []*policyApi.Rule, selectorWithIPsToUpdate map[string][]net.IP, selectorsWithoutIPs []string) error {
+
+			// Prefix is always a /32. Maybe we can just add a /32 to the
+			// prefixes or /128 for IPV4, IPv6?
```

Comment:
Already done, please see `createPrefixLengthCounter()`:

https://github.com/cilium/cilium/blob/9e81aff817bf7a29c7a4782ddbdedc33bc3bace2/daemon/daemon.go#L867-L890

This should also simplify the error paths below.